File: src/main/java/com/learnershi/rclasssocket/entity/StudyData.java
package com.learnershi.rclasssocket.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.Collections;
import java.util.List;

/**
 * StudyData
 *
 */
@Accessors(chain = true)
@NoArgsConstructor
@Getter
@Setter
public class StudyData extends ClassRoomEntity {
    private String name;                // 탭 이름
    private String fileName;        // File 이름

    private int pageIndex;          // 현재 이미지 index
    private List<String> pages;             // 이미지 경로 List
    private List<QuizSubmit> quizSubmit = Collections.emptyList();
}

File: src/main/java/com/learnershi/rclasssocket/entity/StudyData.kt
package com.learnershi.rclasssocket.entity

/**
 * StudyData
 *
 */
class StudyData(
    var name: String? = null,
    var fileName: String? = null,
    var pageIndex: Int = 0,
    var tabIndex: Int = 0,
    var pages: List<String>? = null,
    var quizSubmit: List<QuizSubmit> = emptyList<QuizSubmit>()
) {
    class QuizSubmit {
        private val questionType: String? = null
        private val quizType: String? = null
        private val choices: List<*>? = null
        private val stdName: String? = null
        private val stdAnswer: String? = null
    }
}